### ISSUE TITLE
HTLC database APIs

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-core#1713]
**User Story**
As a user / client developer I want database APIs to query for active HTLC objects.
- [ ] One API to list all HTLC objects.
- [x] APIs to by account, perhaps 2 APIs, one query by `from`, one query by `to`. 
- [x] Pagination should be considered.
- [ ] Perhaps by default order by expiration?
- [x] Perhaps let `get_full_accounts` API return HTLC objects as well.
- [ ] Best have CLI commands as well.

UI work is blocked by this.

**Impacts**
Describe which portion(s) of BitShares Core may be impacted by your request. Please tick at least one box.
- [x] API (the application programming interface)
- [ ] Build (the build process or something prior to compiled code)
- [x] CLI (the command line wallet)
- [ ] Deployment (the deployment process after building such as Docker, Travis, etc.)
- [ ] DEX (the Decentralized EXchange, market engine, etc.)
- [ ] P2P (the peer-to-peer network for transaction/block propagation)
- [ ] Performance (system or user efficiency, etc.)
- [ ] Protocol (the blockchain logic, consensus, validation, etc.)
- [ ] Security (the security of system or user data, etc.)
- [ ] UX (the User Experience)
- [ ] Other (please add below)

**Additional Context (optional)**
Add any other context about your request here.

## CORE TEAM TASK LIST
- [ ] Evaluate / Prioritize Feature Request
- [ ] Refine User Stories / Requirements
- [ ] Define Test Cases
- [ ] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

**Comment by oxarbitrage** [comment]
I am interested on this one. 

Question: Do we want to add the API calls in in database.cpp or do we want to create a new(enabled or disabled by default) `htlc_api` in api.cpp?

**Comment by pmconrad** [comment]
IMO database.cpp, but please use consistent naming.

**Comment by syalon** [comment]
Is it possible to provide an api for query by hashvalue?

**Comment by syalon** [comment]
In addition, preimage length is too unfriendly for users.

**Comment by syalon** [comment]
MobileApp requirements:
* get_full_accounts returns all relevant htlc, including from and to.
* query by hash (if possible)
* preimage length (can omit the best). Is this just for quick verification?

**Comment by abitmore** [comment]
>In addition, preimage length is too unfriendly for users.

Perhaps we can make `preimage_size` effectively optional, e.g. if `size==0`, don't check the size. Need a BSIP.

**Comment by jmjatlanta** [comment]
The majority of these requests have been fixed by #1742. Others are being considered (i.e. sorting), and still others would require a BSIP (i.e. making preimage length optional).

**Comment by abitmore** [comment]
Lack of an API to fetch all HTLCs. See https://github.com/bitshares/bitshares-core/pull/1749#issuecomment-491497694. Moving to next milestone.

**Comment by abitmore** [comment]
Merged #1849. If need a CLI command, create a new issue.
